md/bdm-access-control.md

```diff
@@ -0,0 +1,109 @@
+
+ - **Novice HR**, mapped with the membership 'novice member of human resources' 
+ - **Experimented HR**, mapped with the membership 'experimented member of human resources' 
+
+**III - Create a process and generate some data**
+
+Create a new diagram, with only a start event and an end event.  
+On the pool, add a business variable of type *LeaveRequest*, generate a contract from this data, and generate an instantiation form from this contract.  
+Run this process a couple of time to generate some LeaveRequest data.
+
+**IV - Create a basic living application to display data**
+
+ In the UI Designer,  create an application page (*displayLeaveRequest*):
+
+ - Create a new variable
+	 -  **Name** : leaveRequests
+	 -  **Type** : External API
+	 -  **API URL** :  ../API/bdm/businessData/com.company.model.LeaveRequest?q=find&p=0&c=10
+ 
+ - Add a title to your page (Existing leave requests)
+ - Add a container under the title
+	 - **Collection**: leaveRequests 
+	 - **CSS classes**: alert alert-info
+ - Inside this container, for each of the following attributes of your Business object *(employee - departureDate - duration - medicalComment - managerComment)*:
+	 - Add an input
+		 - Label : [current attribute name]
+		 - Value : $item.[current attribute name]
+		 - Read-only: true
+
+Create a new application descriptor using the [application editor](applicationCreation.md) in the Studio:  
+
+ - Set the application token: leaveRequest
+ - Set the Application Profile: User (ensure that the users mapped with the two profiles created in part II are also mapped with the profile User)
+ - Add an orphan page:
+	 - **Application Page**: custompage_displayLeaveRequest
+	 - **Token**:  displayLeaveRequest
+ - Set the Home page token: displayLeaveRequest
+ - deploy
+
+Ensure that the living application works correctly, and that all the attributes are displayed at the moment.
+
+**V - Define access control for your business object LeaveRequest**
+
+We want to ensure that only experimented member of HR can access to the attribute *medicalComment* of a leave request. A novice member of HR shall not see this attribute.  
+For that, we are going to define two rules for our business object *LeaveRequest*:  
+
+The first rule is the rule for the novices members of HR:
+
+ - **Rule name**: Novice HR Access
+ - **Rule description**: A novice member of HR can't see the medical comment of a leave request
+ - **Attributes checked**:  [employee, departureDate, duration, managerComment]
+ - **Profiles checked**: [Novice HR]
+
+The second rule is the rule for the experimented members of HR:
+
+ - **Rule name**: Experimented HR Access
+ - **Rule description**: An experimented member of HR can see all the attributes of a leave request
+ - **Attributes checked**:  [employee, departureDate, duration, medicalComment, managerComment]
+ - **Profiles checked**: [Experimented HR]
+
+Deploy your access control file
```

Comment:
How the user is supposed to deploy the file?

Reply:
click on the deploy button in the studio